packages/venia-ui/lib/components/Wishlist/WishlistDialog/wishlistDialog.js

```diff
@@ -0,0 +1,117 @@
+import React, { Fragment, useCallback, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import { Relevant } from 'informed';
+
+import { useWishlistDialog } from '@magento/peregrine/lib/talons/Wishlist/WishlistDialog/useWishlistDialog';
+
+import Dialog from '@magento/venia-ui/lib/components/Dialog';
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+import NewWishlistForm from './NewWishlistForm/newWishlistForm';
+import FormError from '../../FormError';
+import WishlistLineItem from './wishlistLineItem';
+
+import defaultClasses from './wishlistDialog.css';
+
+const WishlistDialog = props => {
+    const { isOpen, onClose } = props;
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const talonProps = useWishlistDialog({
+        itemOptions: props.itemOptions,
+        onClose
+    });
+
+    const {
+        canCreateWishlist,
+        formErrors,
+        handleAddToWishlist,
+        handleCancel,
+        handleNewListClick,
+        handleCancelNewList,
+        isAddLoading,
+        isFormOpen,
+        wishlistsData
+    } = talonProps;
+
+    const { formatMessage } = useIntl();
+
+    const createButtonText = formatMessage({
+        id: 'wishlistDialog.createButton',
+        defaultMessage: '+ Create a new list'
+    });
+
+    const maybeListsElement = useMemo(() => {
+        if (wishlistsData) {
+            const wishlists = wishlistsData.customer.wishlists.map(wishlist => {
+                const name = `"${wishlist.name}"`;
+                return (
+                    <Fragment key={wishlist.id}>
+                        <li>
```

Comment:
Do we need the wrapping `Fragment`? Can we just add the `key` to the `<li>` here?

Reply:
Or even include the `<li>` in the `WishlistLineItem` component itself and put the `key` on that?

Reply:
> Do we need the wrapping Fragment? Can we just add the key to the <li> here?

I don't remember why I did this 😢 maybe I had a second element at some point...I'll poke around.

> Or even include the <li> in the WishlistLineItem component itself and put the key on that?

Do we normally put `<li>` in individual components? It wouldn't make sense out of context whereas the item right now may be useable in other places...:shrug:

Reply:
I removed the Fragment.